### ISSUE TITLE
[MPS] torch.baddbmm with beta=0 propagates NaN from input, violating documented semantics

### ISSUE THREAD

**Issue by Bill-Dai1982** [pytorch/pytorch#194442]
### 🐛 Describe the bug

`torch.baddbmm` on MPS propagates NaN from `input` even when `beta=0`. The documentation states:

> If β is 0, then the content of `input` will be ignored, and nan and infinity in it will not be propagated to the output.

On MPS this contract is violated: a single NaN scalar in `input` (shape `(1,1,1)` broadcast) turns the **entire** output into NaN.

### Reproduction

```python
import torch

q = torch.randn(1, 64, 128, device="mps", dtype=torch.bfloat16)
k = torch.randn(1, 128, 3520, device="mps", dtype=torch.bfloat16)

nan_in = torch.full((1, 1, 1), float("nan"), device="mps", dtype=torch.bfloat16)
out = torch.baddbmm(nan_in, q, k, alpha=0.088, beta=0)
print(int(torch.isnan(out).sum()), "/", out.numel())
```

Observed output:

```
225280 / 225280        # MPS: every element is NaN
```

Same code on CPU prints `0`, matching the documented semantics.

Note: with real-world usage the NaN input is not deliberately constructed — it comes from *uninitialized* memory. Any code that initializes the accumulator via `torch.empty(...)` and relies on `beta=0` to ignore it will intermittently read garbage bit patterns that decode to NaN/Inf.

### Real-world impact

ComfyUI's sub-quadratic attention (`comfy/ldm/modules/sub_quadratic_attention.py`) computes attention scores exactly this way:

```python
attn_weights = torch.baddbmm(
    torch.empty(1, 1, 1, device=query.device, dtype=query.dtype),
    query, key_t, alpha=scale, beta=0,
)
```

On Apple Silicon this intermittently poisons whole attention score matrices whenever the recycled memory behind `empty(1,1,1)` contains NaN patterns → diffusion sampling collapses → video models produce all-black outputs at very high rates (~73% of runs observed with LTX-2.x bf16 workflows; deterministic per-seed because allocation layout is deterministic). Initializing with `zeros()` instead of `empty()` eliminates the failures completely.

### Environment

- PyTorch version: 2.13.0
- OS: macOS 26.6.2 (Darwin 25.6.0), arm64
- Device: Apple M5 Max (MPS)
- Python: 3.13.12

Any CPU/CUDA backend honors the documented semantics; MPS appears to multiply by `beta` without special-casing `beta == 0`.

cc @kulinseth @malfet @DenisVieriu97 @jhavukainen @aditvenk @Isalia20

**Comment by Achraf921** [comment]
Opened #194474 for this.

Your bfloat16 case does not reproduce on current main. The MPSGraph path in
addbmm_or_baddbmm_out_mps_impl skips the beta*input node entirely when beta is
0, which landed after 2.13.0, so a nightly should already unblock the ComfyUI
workflows.

Complex is still affected on main, since complex and integral types return
earlier into the Metal kernel which always reads the bias. The PR closes that
gap so the documented behavior holds for every dtype.

**Comment by malfet** [comment]
I've seen a PR fixing something like that a week or two ago

**Comment by Achraf921** [comment]
That's probably #193850, but that one only guards the empty tensor early return. The MPSGraph path has its own beta == 0 check, which is why the bfloat16 case here doesn't reproduce on main anymore. Complex misses both, it returns into do_metal_addbmm_or_baddbmm before either check. That's what I fixed in #194474.
